### PR TITLE
Integrate sdk contract revamping changes

### PR DESCRIPTION
**Pull request by yathindrakodithuwakku**
### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/home.jsp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 <jsp:scriptlet>
     session.setAttribute("authCode", request.getParameter("code"));
     session.setAttribute("sessionState", request.getParameter("session_state"));
+    if(request.getParameter("state") != null) {session.setAttribute("state", request.getParameter("state"));}
```
Comment (yasinmiran) [inline apps/console/src/home.jsp:26]:
```suggestion
    if (request.getParameter("state") != null) {
        session.setAttribute("state", request.getParameter("state"));
    }
```

File: apps/console/src/index.jsp [hunk 2]
```diff
@@ -38,44 +47,37 @@
                                         ? "<%= htmlWebpackPlugin.options.sessionState %>" 
                                         : null;
             
-            function getApiPath(path) {
-                if(path) {
-                    return serverOrigin + path;
-                }
+            if(!authorizationCode) {
```
Comment (yasinmiran) [inline apps/console/src/index.jsp:50]:
```suggestion
            if (!authorizationCode) {
```

File: apps/console/src/index.jsp [hunk 3]
```diff
@@ -38,44 +47,37 @@
...
+                function getApiPath(path) {
+                    if(path) {
```
Comment (yasinmiran) [inline apps/console/src/index.jsp:52]:
```suggestion
                    if (path) {
```

File: apps/console/src/index.jsp [hunk 4]
```diff
@@ -38,44 +47,37 @@
...
+                }
 
-                if(authorizationCode) {
-                    sessionStorage.setItem("userAccessedPath", userAccessedPath.split(window.origin)[1]);
+                var isSilentSignInDisabled = userAccessedPath.includes("disable_silent_sign_in");
+                var isSignOutSuccess = userAccessedPath.includes("sign_out_success");
 
-                    window.location.href = applicationDomain + '/' + "<%= htmlWebpackPlugin.options.basename %>" + '/authenticate?code=' + authorizationCode +
-                                '&session_state='+authSessionState;
+                if(isSignOutSuccess) {
```
Comment (yasinmiran) [inline apps/console/src/index.jsp:75]:
```suggestion
                if (isSignOutSuccess) {
```

File: apps/console/src/index.jsp [hunk 5]
```diff
@@ -38,44 +47,37 @@
...
+                    window.location.href = applicationDomain + '/' + "<%= htmlWebpackPlugin.options.basename %>"
+                }
+                
+                if(isSilentSignInDisabled) {
```
Comment (yasinmiran) [inline apps/console/src/index.jsp:79]:
```suggestion
                if (isSilentSignInDisabled) {
```

File: apps/myaccount/src/home.jsp [hunk 6]
```diff
@@ -27,6 +27,7 @@
 <jsp:scriptlet>
     session.setAttribute("authCode",request.getParameter("code"));
     session.setAttribute("sessionState", request.getParameter("session_state"));
+    if(request.getParameter("state") != null) {session.setAttribute("state", request.getParameter("state"));}
```
Comment (yasinmiran) [inline apps/myaccount/src/home.jsp:30]: format

**Review by yasinmiran — APPROVED** [review]

File: apps/console/src/home.jsp (hunk 1 above)
Comment (yathindrakodithuwakku) [reply apps/console/src/home.jsp:26]: This is intentionally made inline, as this is inside a `<jsp:scriptlet>` tag

File: apps/myaccount/src/home.jsp (hunk 6 above)
Comment (yathindrakodithuwakku) [reply apps/myaccount/src/home.jsp:30]: This is intentionally made inline, as this is inside a `<jsp:scriptlet>` tag

**Pull request merged**